src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -248,7 +249,35 @@ def injectFiles(self):
             myThread.transaction.commit()
 
         return
-
+    
+    def subscribeBlocksInDataset(self, datasetPaths, node, xmlData):
+        """
+        _subscribeBlocksInDataset_
+        :arg list datasetPaths: list of datasetPath
+        :arg str node: site location
+        :arg str xmlData: xml formatted data
+        """
+        
+        subs = []
+
+        # Create the subscription objects and add them to the list
+        # The list takes care of the sorting internally 
+        for datasetPath in datasetPaths:
+            phedexSub = PhEDExSubscription(datasetPath, node, self.group, level = 'block',
+                                           move='n', custodial='n', static='y',
+                                           request_only='n', no_mail='y')
```

Comment:
Hi Seangchan,

I think you want request_only='n' because this will auto-approve the subscription.
no_mail='y' is to suppress the mail notifications.

Cheers
N